src/shogun/multiclass/tree/RelaxedTree.cpp

```diff
@@ -141,7 +141,8 @@ bool CRelaxedTree::train_machine(CFeatures* data)
 	CMulticlassLabels *lab = dynamic_cast<CMulticlassLabels *>(m_labels);
 
 	RelaxedTreeUtil util;
-	SGMatrix<float64_t> conf_mat = util.estimate_confusion_matrix(m_machine_for_confusion_matrix,
+	SGMatrix<float64_t> conf_mat = util.estimate_confusion_matrix(
+			m_machine_for_confusion_matrix->as<CBaseMulticlassMachine>(),
```

Comment:
from compile time safety to runtime safety ... :/

Reply:
we could also add the `CBaseMulticlassMachine` to the shogun  base classes, but I am not sure what we get API wise?
@lisitsyn you wrote that, you have any opinions?